com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/layered/DefaultXtextTargetPlatform.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Group AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Group AG - initial API and implementation
+ *******************************************************************************/
+package com.avaloq.tools.ddk.xtext.layered;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsData;
+
+import com.avaloq.tools.ddk.xtext.build.IBinaryModelStore;
+import com.avaloq.tools.ddk.xtext.resource.IDerivedObjectAssociationsStore;
+import com.avaloq.tools.ddk.xtext.resource.extensions.DelegatingResourceDescriptionsData;
+import com.avaloq.tools.ddk.xtext.resource.extensions.IResourceDescriptionsData;
+
+
+/**
+ * Default implementation implements the legacy mechanism through an empty collection.
+ */
+public class DefaultXtextTargetPlatform implements IXtextTargetPlatform {
+  private volatile IResourceDescriptionsData index; // NOPMD: volatile
+
+  @Override
+  public IResourceDescriptionsData getIResourceDescriptionsData() {
+    if (index == null) {
+      index = new DelegatingResourceDescriptionsData(new ResourceDescriptionsData(Collections.emptyList()));
```

Comment:
@andrewL-avlq, I am using an empty collection instead of an NullPersistedStateProvider, I think it should be enough.